assets/src/dashboard/app/reducer/stories.js

```diff
@@ -42,40 +42,32 @@ function storyReducer(state, action) {
         isLoading: action.payload,
       };
     }
-    case ACTION_TYPES.CLEAR_STORIES_ORDER: {
+    case ACTION_TYPES.FETCH_STORIES_FAILURE:
       return {
         ...state,
-        storiesOrderById: [],
-        stories: {},
-        totalPages: 0,
-        totalStories: 0,
+        isError: action.payload,
       };
-    }
-    case ACTION_TYPES.UPDATE_STORIES:
-      // eslint-disable-next-line no-case-declarations
-      let uniqueStoryIds = [
-        ...state.storiesOrderById,
-        ...action.payload.map(({ id }) => id),
-      ];
-      uniqueStoryIds = uniqueStoryIds.filter((value, index, self) => {
+
+    case ACTION_TYPES.FETCH_STORIES_SUCCESS: {
+      const fetchedStoriesById = action.payload.stories.map(({ id }) => id);
+      const combinedStoryIds =
+        action.payload.page === 1
+          ? fetchedStoriesById
+          : [...state.storiesOrderById, ...fetchedStoriesById];
+
+      const uniqueStoryIds = combinedStoryIds.filter((value, index, self) => {
```

Comment:
I'm a little confused at the logic here. Perhaps some comments around the map and why combining IDs and the filter.

Reply:
totally fair, it's a little weird. Basically, doing this to maintain the order of pagination while actually doing scroll. Most of the time it's unnecessary but the IDs will duplicate.